smart-py/marketplace.py

```diff
@@ -1,64 +1,203 @@
+import smartpy as sp
+
 class Marketplace(sp.Contract):
     def __init__(self, objkt, metadata, manager, fee):
         self.init(
             objkt = objkt,
             metadata = metadata,
             manager = manager,
-            swaps = sp.big_map(tkey=sp.TNat, tvalue=sp.TRecord(issuer=sp.TAddress, objkt_amount=sp.TNat, objkt_id=sp.TNat, xtz_per_objkt=sp.TMutez, royalties=sp.TNat, creator=sp.TAddress)),
+            swaps = sp.big_map(
+                tkey=sp.TNat,
+                tvalue=sp.TRecord(
+                    issuer=sp.TAddress,
+                    objkt_amount=sp.TNat,
+                    objkt_id=sp.TNat,
+                    xtz_per_objkt=sp.TMutez,
+                    royalties=sp.TNat,
+                    creator=sp.TAddress
+                )
+            ),
             counter = 500000,
             fee = fee
-            )
-            
+        )
+
     @sp.entry_point
     def swap(self, params):
-        sp.verify((params.objkt_amount > 0) & ((params.royalties >= 0) & (params.royalties <= 250)))
-        self.fa2_transfer(self.data.objkt, sp.sender, sp.self_address, params.objkt_id, params.objkt_amount)
-        self.data.swaps[self.data.counter] = sp.record(issuer=sp.sender, objkt_amount=params.objkt_amount, objkt_id=params.objkt_id, xtz_per_objkt=params.xtz_per_objkt, royalties=params.royalties, creator=params.creator)
+        # previous for easy comparison
+        # sp.verify((params.objkt_amount > 0) & ((params.royalties >= 0) & (params.royalties <= 250)))
+
+        sp.verify(
+            (params.objkt_amount > 0) &
+            (params.objkt_id < self.data.counter) &
+            (params.objkt_id >= 152) &
+            (
+                (params.royalties >= 0) & (params.royalties <= 250)
+            )
+        )
+
+        # TODO
+        # i want to validate the royalties here to make sure they match
+        # what is set on the objkt when it is minted
+
+        # previous for easy comparison
```

Comment:
Same as above.

Reply:
I would also find much easier to read if there is a verify per condition
sp.verify(params.objkt_amount > 0))
sp.verify(params.royalties >= 0)
sp.verify(params.royalties <= 250)

Reply:
100% agreed

Reply:
how much of https://smartpy.io/ide?cid=QmPMQ12VXeJSCzYEmzBsDL3DtW29y1aF7Y7phJB8fdPEwH&k=f9af4cbf831286a22f4b is boilerplate?